File: app/src/main/java/com/practice/forecast/ui/detail/DetailContract.java
package com.practice.forecast.ui.detail;

import com.practice.forecast.ui.arch.Contract;
import com.practice.forecast.ui.arch.mvi.ScreenState;
import com.practice.weathermodel.pojo.City;

import java.util.List;

import androidx.lifecycle.LiveData;

public interface DetailContract {

    interface BaseDetailViewModel {

        void downloadCity(String cityId);

        LiveData<List<City>> getCityObservable();
        LiveData<DetailScreenState> getStateHolderObservable();
    }

    interface View extends Contract.View {
        void showProgress();
        void hideProgress();
        void setListToAdapter(List<City> weatherLIst);
        void showError();
    }

    interface Host extends Contract.Host {
        void backToMapFragment();

    }
}

File: app/src/main/java/com/practice/forecast/ui/detail/DetailContract.kt
package com.practice.forecast.ui.detail

import androidx.lifecycle.Lifecycle
import androidx.lifecycle.LifecycleObserver
import androidx.lifecycle.LiveData
import com.practice.forecast.ui.arch.Contract
import com.practice.weathermodel.pojo.City

interface DetailContract {
    interface BaseDetailViewModel: LifecycleObserver {
        fun downloadCity(cityId: String?)
        fun getCityObservable(): LiveData<List<City>>
        fun getStateHolderObservable(): LiveData<DetailScreenState?>
    }

    interface View : Contract.View {
        fun showProgress()
        fun hideProgress()
        fun setListToAdapter(weatherLIst: List<City?>?)
        fun showError()
    }

    interface Host : Contract.Host {
        fun backToMapFragment()
    }
}